### PR TITLE
distsql: support global index in `memIndexLookUpReader`

### LINKED ISSUE #52132: The global index not work with `memIndexLookUpReader` executor
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
set tidb_enable_global_index=true;
CREATE TABLE `t` (
  `a` int(11) DEFAULT NULL,
  `b` int(11) DEFAULT NULL,
  UNIQUE KEY `idx1` (`b`),
  KEY `idx` (`b`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
PARTITION BY HASH (`a`) PARTITIONS 5;

begin;
insert into t values (1, 2), (2, 3);
select b from t use index(idx1) where b > 2;
```

### 2. What did you expect to see? (Required)

return result `3`

### 3. What did you see instead (Required)

return empty

### PR DESCRIPTION
**Pull request by Defined2014**
### What problem does this PR solve?

Issue Number: close #52132

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/tablecodec/tablecodec.go [hunk 1]
```diff
@@ -975,20 +975,33 @@ func decodeHandleInIndexKey(keySuffix []byte) (kv.Handle, error) {
...
 
-func decodeHandleInIndexValue(value []byte) (kv.Handle, error) {
+func decodeHandleInIndexValue(value []byte) (handle kv.Handle, err error) {
+	var seg IndexValueSegments
 	if getIndexVersion(value) == 1 {
-		seg := SplitIndexValueForClusteredIndexVersion1(value)
-		return kv.NewCommonHandle(seg.CommonHandle)
+		seg = SplitIndexValueForClusteredIndexVersion1(value)
+	} else {
+		if len(value) <= MaxOldEncodeValueLen {
```
Comment (zimulala) [inline pkg/tablecodec/tablecodec.go:983]: Maybe `getIndexVersion(value) == 0` is more readable, Or you can set a temporary variable for "getIndexVersion". Mainly to unify with line980.

**Review by zimulala — APPROVED** [review]
LGTM

File: pkg/tablecodec/tablecodec.go (hunk 1 above)
Comment (Defined2014) [reply pkg/tablecodec/tablecodec.go:982]: Comments updated.

**Review by tiancaiamao — APPROVED** [review]

**Review by XuHuaiyu — APPROVED** [review]

**Pull request merged**